### ISSUE TITLE
 Live search does not announce results to assistive technologies

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15294]
Live search does not announce results to assistive technologies

-   **Severity:** Medium
-   **Affected Populations:**
    -   Blind
    -   Low-Vision
    -   Cognitively Impaired
-   **Platform(s):**
    -   Windows - Screen Reader
    -   Mac - VoiceOver
    -   Android - TalkBack
    -   iOS - VoiceOver
-   **Components affected:**
    -   Media Dialog

#### Issue description

When users in the "Featured Image" modal's "Media Library" do a
search using the search/filter control, the results change dynamically,
however these changes are not announced to screen readers, so blind and
low-vision users cannot tell what, if anything, matches the search
without manually reading through the image list.

Assistive technologies rely on state information to understand the
behaviour of dynamic content, and if this information is incomplete or
inaccurate it may create a confusing experience for users.

##### Issue Code

```
    <div class="media-toolbar-primary search-form">


        <label for="media-search-input" class="screen-reader-text">Search Media</label>


        <input type="search" placeholder="Search media items..." id="media-search-input" class="search">


    </div>
```

#### Remediation Guidance

Use a live region to announce the number of matched results, updating as
users type and delete characters (ideally with a short timeout or
throttling so that the announcements don't interfere with user typing).

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-63 in Tenon's report

**Comment by afercia** [comment]
Worth noting the media modals aren't part of Gutenberg. This issue should be moved to the core Trac under the Media component.

**Comment by melchoyce** [comment]
Full report screenshot:

![image](https://user-images.githubusercontent.com/2846578/57186783-46999500-6e9a-11e9-8473-8f8d18e486b2.png)

**Comment by anevins12** [comment]
Moved to Trac: https://core.trac.wordpress.org/attachment/ticket/47146/

**Comment by afercia** [comment]
Closing as moved to Trac.
